src/multivariate/multinomial.jl

```diff
@@ -140,21 +140,20 @@ function insupport(d::Multinomial, x::AbstractVector{T}) where T<:Real
     return s == ntrials(d)  # integer computation would not yield truncation errors
 end
 
-function _logpdf(d::Multinomial, x::AbstractVector{T}) where T<:Real
+function _logpdf{T<:Real}(d::Multinomial, x::AbstractVector{T})
     p = probs(d)
     n = ntrials(d)
     S = eltype(p)
     R = promote_type(T, S)
+    insupport(d,x) || return -R(Inf)
     s = R(lgamma(n + 1))
-    t = zero(T)
     for i = 1:length(p)
         @inbounds xi = x[i]
         @inbounds p_i = p[i]
-        t += xi
-        s -= R(lgamma(xi + 1))
-        @inbounds s += xi * log(p_i)
-    end
-    return ifelse(t == n, s, -R(Inf))
+        s -= R(lgamma(R(xi) + 1))
+        @inbounds s += ifelse((xi==0) & (p_i==0), 0.0, xi * log(p_i)) # log(0^0)=0 not NaN
```

Comment:
Do you see a performance improvement with `@inbounds` here? Doesn't look like there's any indexing happening in this step, so it shouldn't be necessary. But if you want you could move the `@inbounds` annotation from in front of the variable definitions to right before `for`.

Reply:
Also for type stability should this be `zero(T)` rather than the `Float64` literal 0.0?

Reply:
1) I just edited the maths leaving the macros as in master.
2) you are right, edited for commit.